### PR TITLE
SALTO-5109: Salesforce Metadata Important Values

### PR DESCRIPTION
**Pull request by tamtamirr**
Add initial implementation for Salesforce Important Values for Metadata Types.

---

- This feature is disabled by default. In order to enable it set the `importantValues` optional feature to true in the adapter config.
- Add the ability to configure additional important values under the adapter `fetch` config. If `important value` definition is already set for the `value`, we prefer the one from the `additional important values`.

Workspace diff: https://github.com/salto-io/tamir-sf/commit/6533cfebf854bd1a663bd825941e8a14346c49d9

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/types.ts [hunk 1]
```diff
@@ -798,6 +802,10 @@ const fetchConfigType = createMatchingObjectType<FetchParameters>({
     preferActiveFlowVersions: { refType: BuiltinTypes.BOOLEAN },
     addNamespacePrefixToFullName: { refType: BuiltinTypes.BOOLEAN },
     warningSettings: { refType: warningSettingsType },
+    additionalImportantValues: {
+      // Exported type is downcasted to TypeElement
+      refType: CoreAnnotationTypes[CORE_ANNOTATIONS.IMPORTANT_VALUES] as ListType<ObjectType>,
+    },
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/types.ts:808]: Do we actually want a list of `ObjectType` in `salesforce.nacl`? Should this be a list of names or references or something?

File: packages/salesforce-adapter/src/filters/important_values.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      const additionalImportantValues = makeArray(config.fetchProfile.additionalImportantValues)
+      if (additionalImportantValues.length === 0) {
+        return DEFAULT_IMPORTANT_VALUES
+      }
+      // The additional important values should override the default ones
+      const additionalImportantValuesValues = new Set(additionalImportantValues.map(value => value.value))
+      return DEFAULT_IMPORTANT_VALUES
+        .filter(importantValue => !additionalImportantValuesValues.has(importantValue.value))
+        .concat(additionalImportantValues)
+    }
```
Comment (tomsellek) [inline packages/salesforce-adapter/src/filters/important_values.ts:71]: Why not move this into the fetch profile? Perhaps create a method that takes a type+field and returns whether it's an important value?
Or maybe this is a fit for `MetadataQuery` instead?

Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/important_values.ts:71]: I think having all the logic that is relevant to `improtant values` with this feature is less complex for code readers.
In fact my initial implementation was this code within the `Fetch Profile` but it felt forgein there.
I also think that `FetchProfile` should only wrap the `fetch / deploy` settings (Yes, it's name is misleading) and not have more complex logic. WDYT?

File: packages/salesforce-adapter/src/types.ts (hunk 1 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/types.ts:808]: Was completely tired when I wrote this :monkey: Thanks!

File: packages/adapter-api/src/builtins.ts [hunk 3]
```diff
@@ -103,7 +103,7 @@ const restrictionType = new ObjectType({
   },
 })
 
-const importantValueType = new ObjectType({
+export const importantValueType = new ObjectType({
```
Comment (tamtamirr) [inline packages/adapter-api/src/builtins.ts:106]: @edenhassid are we okay with export? 🙏

Comment (edenhassid) [reply packages/adapter-api/src/builtins.ts:106]: yes

File: packages/salesforce-adapter/src/filters/important_values.ts (hunk 2 above)
Comment (tomsellek) [reply packages/salesforce-adapter/src/filters/important_values.ts:71]: Personally, I like the abstraction of e.g. `MetadataQuery` as means of encapsulating the specifics of the config file structure, and ensuring the rest of the code doesn't need to know about them.

Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/important_values.ts]: I've done a refactor 🙏

**Review by tomsellek — APPROVED** [review]
:)

**Pull request merged**